File: engine/b/src/main/java/net/crazyinvestor/engine_b/repository/TickerRepository.java
package net.crazyinvestor.engine_b.repository;

import net.crazyinvestor.engine_b.entity.Symbol;
import net.crazyinvestor.engine_b.entity.Ticker;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TickerRepository extends JpaRepository<Ticker, Long> {
    Optional<Ticker> findBySymbolAndTickType(Symbol symbol, String tickType);
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/repository/TickerRepository.kt
package net.crazyinvestor.ticker_producer_engine.repository

import net.crazyinvestor.ticker_producer_engine.entity.Symbol
import net.crazyinvestor.ticker_producer_engine.entity.Ticker

import org.springframework.stereotype.Repository
import java.util.*

@Repository
class TickerRepository {
    fun findBySymbolAndTickType(symbol: Symbol, tickType: String): Optional<Ticker> {
        println("MOCK REPOSITORY5 $symbol $tickType")

        return Optional.of(Ticker(null,symbol, tickType))
    }

    fun save(ticker: Ticker): Ticker {
        println("save() MOCK REPOSITORY5 $ticker")
        return ticker
    }
}